### ISSUE TITLE
RemoveEdge incorrectly throws error in specific case 

### ISSUE THREAD

**Issue by Smuzzy-waiii** [acmpesuecc/Onyx#1]
The `RemoveEgde` function throws "Key not found" error while trying to remove an edge `a->b` if there exist no edges with `a` as their source. This is happening due to the following code:
```go
	item, err := txn.Get([]byte(from))
	if err != nil {
		return err
	}
```
where we are not checking if the error returned by badger is `ErrKeyNotFound` as is being done for the `AddEdge` function

------------------------------------------

Modify the `RemoveEdge` function to return `(bool, error)` instead of just a `error` where `bool` signifies whether the Remove actually removed an edge or whether it silently failed because the edge didn't exist. 
Have the function return:
- `true, nil` on successful removal of an edge that exists
- `false, nil` if the edge doesn't exist
- `false, error` if there is any error

Note that you will have to check for the edge not existing is 2 places: while getting the edgelist for the src node from badger, and then again in the edgelist before calling `delete(dstNodes, to)`

**You are also required to add a unit test to cover this case**

*The above change is a breaking change to the `RemoveEdge` API. Make sure to edit all the calls to `RemoveEdge` in `lib_test.go` to reflect that*

**Comment by JP-sDEV** [comment]
Hi @Smuzzy-waiii, could I be assigned this bug?

**Comment by Smuzzy-waiii** [comment]
@JP-sDEV This repo is being prepped for a university club event called Hacknight where participants get to solve issues on FOSS repo's to get points. You can find out more about hacknight on the ACM PESU ECC instagram [here](https://www.instagram.com/acmpesuecc). The event is from 4PM IST on Fri, 18th Oct  to 12PM IST on Sat, 19th Oct. 

If this issue is still unresolved at the end of the event, I would love to assign it to you

**Comment by JP-sDEV** [comment]
No worries, thanks!

**Comment by velukutty2194** [comment]
could i be assigned??

**Comment by Smuzzy-waiii** [comment]
!assign @velukutty2194

**Comment by Smuzzy-waiii** [comment]
The bot will deassign you in 45mins. If you need more time, show your progress and I can extend the time. Please feel free to text/call me at 8618950413

**Comment by Smuzzy-waiii** [comment]
@velukutty2194 Are you still working on this?

**Comment by Smuzzy-waiii** [comment]
!extend

**Comment by Smuzzy-waiii** [comment]
!deassign

**Comment by Smuzzy-waiii** [comment]
@JP-sDEV Are you still interested in working on this? I can assign you

**Comment by JP-sDEV** [comment]
@Smuzzy-waiii  Yes, I am still interested! I will start working on it on Monday, if assigned.

**Comment by Smuzzy-waiii** [comment]
@JP-sDEV Yes np go for it! 😃
